foundation/foundation-mda/src/main/java/com/boozallen/aiops/mda/generator/SparkApplicationGenerator.java

```diff
@@ -143,6 +145,21 @@ public void generate(GenerationContext context) {
         vc.put(VelocityProperty.PIPELINE, pipeline.getName());
         vc.put(VelocityProperty.DOCKER_PROJECT_REPOSITORY_URL, dockerProjectRepositoryUrl);
 
+        /*
+         * Autogenerate the environment variables when a filestore is declared for the pipeline
+         */
+        List<FileStore> fileStores = pipeline.getFileStores();
+        boolean hasFileStore = fileStores.isEmpty();
+        if(!hasFileStore){
+            List<String> fileStoresDecorated = new ArrayList<>();
+            for(FileStore fileStore : fileStores){
+                BaseFileStoreDecorator decoratedStore = new BaseFileStoreDecorator(fileStore);
+                fileStoresDecorated.add(decoratedStore.getName());
+            }
+            vc.put(VelocityProperty.FILE_STORE_NAMES, fileStoresDecorated);
```

Comment:
Q: Why not just pass the actual decorated file stores and let the template pull the name? This would allow us to more easily extend file store objects to hold more information.

Reply:
Refactored.